### ISSUE TITLE
HSTS header not getting sent on box virtualhost on /admin or Z-Push

### ISSUE THREAD

**Issue by JoshHarmon** [mail-in-a-box/mailinabox#878]
I'm not seeing the HSTS header get set on the /admin URI of the box virtualhost. It does appear to be sent on all the others (other domains are okay; /mail sends it).

According to Postman (Chrome app), the following headers were sent when I made a GET request to the /admin URI of box.mydomain.com:

```
Connection →keep-alive
Content-Encoding →gzip
Content-Security-Policy →frame-ancestors 'none';
Content-Type →text/html; charset=utf-8
Date →Tue, 28 Jun 2016 21:31:09 GMT
Server →nginx
Transfer-Encoding →chunked
X-Content-Type-Options →nosniff
X-Frame-Options →DENY
```

**Comment by yodax** [comment]
I have confirmed it is applied to box.domain.tld but not in the headers for the sub admin. Cloud does send it, also custom (html) site send the header. I will look at a patch.

**Comment by yodax** [comment]
Z-push doesn't send it either. Shall we do that one as well? I don't know if that would break anything since z-push/active sync is so incredibly picky.

**Comment by JoshHarmon** [comment]
@yodax I've never once gotten Z-Push to keep work correctly, so I don't have any particularly great way to help test that. I do think that we should be sending HSTS wherever possible though.

**Comment by yodax** [comment]
I'm not sure that adding the header to z-push would help in adding security. If you add a new account you can accept (on iOS) a self signed certificate. I don't know that it will remember the header if you remove the account. Also if the account is created with https, it will not suddenly connect over http.

Given the "unreliable" / error prone state of z-push i'm hesitant to  try this out.

**Comment by JoshData** [comment]
Closed via #879. I think we can ignore Z-Push.
